src/clang_utils.cc

```diff
@@ -20,30 +20,49 @@ lsRange GetLsRangeForFixIt(const CXSourceRange& range) {
 
 }  // namespace
 
+// See clang_formatDiagnostic
 optional<lsDiagnostic> BuildAndDisposeDiagnostic(CXDiagnostic diagnostic,
                                                  const std::string& path) {
   // Get diagnostic location.
   CXFile file;
-  unsigned int line, column;
-  clang_getSpellingLocation(clang_getDiagnosticLocation(diagnostic), &file,
-                            &line, &column, nullptr);
+  unsigned start_line, start_column, end_line, end_column;
+  clang_getSpellingLocation(clang_getDiagnosticLocation(diagnostic),
+                            &file, &start_line, &start_column, nullptr);
 
-  // Only report diagnostics in the same file. Using
+  // TODO Only report diagnostics in the same file. Using
   // clang_Location_isInSystemHeader causes crashes for some reason.
-  if (path != FileName(file)) {
+  if (file && path != FileName(file)) {
```

Comment:
Should this be clang_isNull or some similar API function instead of if (file ...)?

Reply:
file is a `void *`